packages/core/src/new-api/internal/module-builder.ts

```diff
@@ -364,6 +367,52 @@ export class IgnitionModuleBuilderImplementation<
     return future;
   }
 
+  public readEventArgument(
+    futureToReadFrom:
+      | NamedContractDeploymentFuture<string>
+      | ArtifactContractDeploymentFuture
+      | NamedContractCallFuture<string, string>,
+    eventName: string,
+    argumentName: string,
```

Comment:
what if a user wants to read more than one argument from the event?

Reply:
I'd say they create two features for now. If this becomes a recurrent complain we extend this api.